packages/adp-tooling/test/unit/writer/index.test.ts

```diff
@@ -87,4 +90,18 @@ describe('ADP writer', () => {
             ).toMatchSnapshot();
         });
     });
+
+    describe('migrate', () => {
```

Comment:
The test is fully correct, however, I'd like to slightly simplify it so that it is easier to understand and enhance it in the future.

I would suggest to remove all changes from the root describe, and just add the more concise 

```typescript
    describe('migrate', () => {
        const migrateInputDir = join(__dirname, '../../fixtures/webide-adaptation-project');

        test('migrate minimal config', async () => {
            const projectDir = join(outputDir, 'webide-adaptation-project');
            fs.copy(migrateInputDir, projectDir, { globOptions: { dot: true } });

            await migrate(projectDir, config, fs);
            expect(fs.dump(projectDir)).toMatchSnapshot();
            expect(fs.exists(join(projectDir, '.che/project.json'))).toBeFalsy();
            expect(fs.exists(join(projectDir, 'neo-app.json'))).toBeFalsy();
        });
    });
```
Reasoning:
* everything that is only relevant for migration is in the migration describe block
* structure and names (e.g. `projectDir`) is exactly like it is done in generate
* the expect calls before the actual test are removed because "we trust" our test setup
* remove the two checks of existing files because these two are checked with the snapshot (note: I kept the check for the deleted files because they don't show up in the snapshot since they are added and deleted on the same editor instance)

Reply:
Thank you for the suggestion and the thorough explanation, appreciate it. I have added the changes.